File: Webpage/server/webpage.server/webpage.server/src/main/java/abbruzzese/webpage/webpage/server/webpage/Repository/EventImageRepository.java
package abbruzzese.webpage.webpage.server.webpage.Repository;

import java.util.List;
import java.util.Optional;
import org.springframework.data.repository.CrudRepository;
import abbruzzese.webpage.webpage.server.webpage.entities.EventImage;

public interface EventImageRepository extends CrudRepository<EventImage,String> {

    List<EventImage> findByState(String state);

    Optional<EventImage> findByEventImage(String eventImage);

}

File: PHP_API/REPOSITORY/EventImageRepository.php
<?php 

namespace App\Repository;

use Doctrine\ORM\EntityRepository;
use App\Entity\EventImage;

class EventImageRepository extends EntityRepository
{
    public function findByState(string $state): array
    {
        return $this->findBy(['estado' => $state]);
    }

    public function findByEventImage(string $eventImage): ?EventImage
    {
        return $this->findOneBy(['eventimage' => $eventImage]);
    }

}

?>